### DISCUSSION TITLE
1.9

### DISCUSSION THREAD

**Discussion by kirb** [LegacyUpdate/LegacyUpdate#204] [Announcements]
> [!IMPORTANT]
> If nothing happens when you open Legacy Update setup on Windows XP, [follow these steps](https://legacyupdate.net/faq/issues) to work around a Windows bug.

* Legacy Update can now update itself when you use it. If you already have Legacy Update installed, just visit the website in Internet Explorer to update to 1.9!
* Adds support for installing Service Pack 3 of the original 2001 edition of Windows XP Embedded. (Contributed by @renodr - thanks!)
* Adds support for installing Update Rollup 4 of Windows Home Server 2011. (Contributed by @renodr and @stdin82 - thanks!)
* Fixes Legacy Update setup crashing when it attempts to enable Microsoft Update on Windows 7.
* Fixes Certificate Trust List being unnecessarily redownloaded after Legacy Update restarts the computer.
* General internal cleanup of the Legacy Update setup code.

<hr /><em>This discussion was created from the release <a href='https://github.com/kirb/LegacyUpdate/releases/tag/v1.9.0.0'>1.9</a>.</em>

**Comment by kristibektashi** [comment]
Perhaps the Windows XP issue could be bypassed by signing the setup file with both an SHA1 and an SHA256 certificate. That way Windows XP will be able to recognize the SHA1 certificate, while still verifying the authenticity of the file with the SHA256 certificate.

**Reply by kirb** [reply]
It’s already signed by me with dual sha1/sha256 hashes, but the certificate chain is fully sha256, and I believe the root is not trusted by a stock install of XP SP2/SP3 (it is once you update root certs in Legacy Update). I don’t think there's really a solution around this, other than perhaps asking Certum whether they still provide a legacy XP-compatible chain we can use. At worst, I might need to spend $300+ again at another CA who does support this, but I think I’d rather wait for the cert to expire to do that.
